src/base/PoolManagerOwnable2Step.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {IPoolManagerOwner} from "../interfaces/IPoolManagerOwner.sol";
+
+/**
+ * @notice dev This contract implements "Ownable2Step styled" poolManager ownership transfer
+ * functionality. Namely an extra acceptance step is added to the ownership transfer process.
+ * This is done to prevent accidental ownership transfer to a wrong address.
+ */
+abstract contract PoolManagerOwnable2Step is IPoolManagerOwner {
+    error NotPendingPoolManagerOwner();
+
+    address private _pendingPoolManagerOwner;
```

Comment:
why not just make it public?

Reply:
loll, actually refer this from OZ@Ownable2Step

Reply:
> why not just make it public?

Public increases the bytecode size since there is now a public view function for it that can be called.